agent/api/task/task.go

```diff
@@ -696,13 +707,33 @@ func (task *Task) BuildCNIConfig() (*ecscni.Config, error) {
 			break
 		}
 	}
-
 	// If there is ipv6 assigned to eni then set it
 	if len(eni.IPV6Addresses) > 0 {
 		cfg.ENIIPV6Address = eni.IPV6Addresses[0].Address
 	}
+}
+
+// convertAppMeshToCNIConfig converts input app mesh config into cni config
+func convertAppMeshToCNIConfig(appMesh *apiappmesh.AppMesh, cfg *ecscni.Config) {
+	cfg.AppMeshCNIEnabled = appMesh.AppMeshEnabled
+	cfg.IgnoredUID = appMesh.IgnoredUID
+	cfg.IgnoredGID = appMesh.IgnoredGID
+	cfg.ProxyIngressPort = appMesh.ProxyIngressPort
+	cfg.ProxyEgressPort = appMesh.ProxyEgressPort
+	cfg.AppPorts = appMesh.AppPorts
+
+	var egressIgnoredIPs []net.IP
+	for _, ip := range appMesh.EgressIgnoredIPs {
+		ip := net.ParseIP(ip)
+		if ip == nil {
+			seelog.Warnf("app mesh config: malformed egress ignored ip")
```

Comment:
If I understand correctly, if the ip is invalid, we just ignore it and log as warning? If this is the case, I am not sure whether this is the right action since it is hard for customers to figure out what happened if only some of ip settings are missing. We need to discuss about this. Also, the log message needs to have the malformed ip printed out so we can know which one causes it.

Reply:
ditto